presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java

```diff
@@ -89,6 +92,18 @@ public RuleAssert withSession(Session session)
         return this;
     }
 
+    public RuleAssert withStats(String nodeId, PlanNodeStatsEstimate nodeStats)
+    {
+        ((TestingStatsCalculator) statsCalculator).stats.put(new PlanNodeId(nodeId), nodeStats);
```

Comment:
please provide some method instead of accessing `stats` map directly

Reply:
why?

Reply:
maybe check first that we can do the cast since it's not guaranteed

Reply:
What check message you would want to see here, so that this is more useful for a developer writing a test, than just CCE?

Reply:
I guess cce is pretty clear.  But maybe some comment on the method that it should only be used if your StatsCalculator is a TestingStatsCalculator so it's easy for people to notice that it's not general purpose would be helpful.  The either/or requirement for `withStats` vs. `withStatsCalculator` would probably be unexpected for someone without context writing a new test, so I'm trying to think about how to mitigate that, at least in words if it can't be fixed in the code.